src/auth/src/token.rs

```diff
@@ -36,3 +38,9 @@ pub struct Token {
     /// This might include information like granted scopes or other claims.
     pub metadata: Option<std::collections::HashMap<String, String>>,
 }
+
+#[async_trait::async_trait]
+#[allow(dead_code)] // TODO(#442) - implementation in progress
+pub(crate) trait TokenProvider: Send + Sync {
```

Comment:
`pub(crate)` is fine for now.  You folks will need to decide if external applications can implement their own `TokenProvider`.  If you do, they this should be `pub` instead of `pub(create)`.

Reply:
Ack. I don't think we want the thing public. If we change our minds, we can always go private -> public later.